### PR TITLE
Feature add view page ExpectedHours

### PR DESCRIPTION
**Pull request by Merrique**

### CODE REVIEW

File: managers/tests/test_views.py [hunk 1]
```diff
@@ -342,3 +344,73 @@ def test_remove_task_activities_view_should_change_task_status_to_not_active(sel
...
+        self.url = reverse(
+            "project-expected-hours",
+            kwargs={"pk": self.project.pk, "year": self.current_time.year, "month": self.current_time.month},
+        )
+
+    def test_project_expected_hours_view_should_display_expected_hours_for_project_on_get(self):
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, ProjectExpectedHoursView.template_name)
+
```
Comment (marsolekk) [inline managers/tests/test_views.py:378]: I think that you should also create expected hours for user and check that proper data are displayed in template

Comment (marsolekk) [reply managers/tests/test_views.py:378]: Please also add test that to check that only authenticated users are able to view this page. I am not sure how this should work, bu also only manager/superuser should be able to view this page if I am not wrong, so please also add test for that.

Comment (marsolekk) [reply managers/tests/test_views.py:378]: Maybe this is not related now, but @kbeker what do you think to add another test for removing user from project case? Should "archive" expected hours still display in template?

**Review by marsolekk — CHANGES_REQUESTED** [review]
I found some issues. Please address my comments. 
Also, name of branch is invalid. PR name is strongly related with branch name, as it is feature, branch name should start with `feature/` not `chore/`. I think that you should probably change name of branch (this will automatically close that PR, so maybe wait for @kbeker opinion if he would accept that mistake)

File: managers/templates/managers/partials/create_expected_hours.html [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                <button type="button" class="btn btn-default modal-button" data-dismiss="modal">{{ UI_text.CLOSE_BUTTON.value }}</button>
+            </div>
+            <div class="col-md-6 add-report-form-button">
+                <button type="button" class="submit-btn btn btn-primary modal-button">{{ UI_text.ADD_BUTTON.value }}</button>
+            </div>
+       </div>
+    </div>
+</form>
+
+<script src="{% static '/bootstrap_datepicker_plus/js/datepicker-widget.js' %}"></script>
```
[the remark this answers is not in the document]
Comment (Merrique) [reply managers/templates/managers/partials/create_expected_hours.html:76]: Added possibility to remove and update expected hours.

File: managers/templates/managers/partials/project_expected_hours_per_user.html [hunk 3]
```diff
@@ -0,0 +1,38 @@
+{% load data_structure_element_selectors %}
+{% load data_display_filters %}
+
+<table class="table">
+    <thead>
+        <tr>
+            <td class="td-text-align-left-name" colspan="1">
+                <h4><strong>{{ UI_text.EXPECTED_HOURS_SUMMARY_HEADER.value }}</strong></h4>
```
[the remark this answers is not in the document]
Comment (Merrique) [reply managers/templates/managers/partials/project_expected_hours_per_user.html:8]: Changed strings

File: managers/forms.py [hunk 4]
```diff
@@ -67,3 +72,23 @@ class Meta:
     def save(self, commit: bool = True) -> Project:
         project = super().save()
         return project
+
+
+class ExpectedHoursForm(BSModalModelForm):
+    class Meta:
+        model = ExpectedHours
+        fields = ("date", "user", "project", "value")
```
[the remark this answers is not in the document]
Comment (Merrique) [reply managers/forms.py:80]: Changed

File: managers/templates/managers/partials/expected_hours_for_date_for_project_for_user.html [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+{% regroup user.list by date as expected_hours_by_date %}
+    {% for date in expected_hours_by_date %}
+        {% for expected_hours in date.list %}
+            <tr class="tr-next-day-separator">
+                <td colspan="2"><strong>{{ expected_hours.date }}</strong></td>
+                <td colspan="2">{{ expected_hours.value|duration_field_to_string }}</td>
+            </tr>
+        {% endfor %}
+    {% endfor %}
```
[the remark this answers is not in the document]
Comment (Merrique) [reply managers/templates/managers/partials/expected_hours_for_date_for_project_for_user.html:11]: Added blank line. changed name to ` expected_hours_in_date_for_user.html`.

File: managers/tests/test_views.py (hunk 1 above)
Comment (Merrique) [reply managers/tests/test_views.py:378]: 1) Added new view/page that displays expected hours for user. 
2) Added tests

**Comment by Merrique** [comment]
Closing as work will be continued on a different branch.

**Comment by kbeker** [comment]
Opened with a different branch name https://github.com/Code-Poets/sheetstorm/pull/612

**Pull request closed without merging**